components/Tabs/tests/unit/MDCTabBarRippleTests.m

```diff
@@ -0,0 +1,193 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MDCItemBar.h"
+#import "MDCItemBarCell.h"
+#import "MDCItemBarStyle.h"
+#import "MaterialInk.h"
+#import "MaterialRipple.h"
+#import "MaterialTabs.h"
+
+@interface MDCTabBar (Testing)
+- (void)updateItemBarStyle;
+@end
+
+@interface MDCItemBarCell (Testing)
+@property(nonatomic, strong) MDCRippleTouchController *rippleTouchController;
+@property(nonatomic, strong) MDCInkTouchController *inkTouchController;
+@end
+
+@interface MDCItemBar (Testing)
+@property(nonatomic, strong, nullable) MDCItemBarStyle *style;
+@property(nonatomic, strong, nullable) UICollectionView *collectionView;
+@end
+
+/**
+ This class confirms behavior of @c MDCTabBar when used with Ripple.
+ */
+@interface MDCTabBarRippleTests : XCTestCase
+
+@property(nonatomic, strong, nullable) MDCTabBar *tabBar;
+@property(nonatomic, strong, nullable) MDCItemBar *itemBar;
+
+@end
+
+@implementation MDCTabBarRippleTests
+
+- (void)setUp {
+  [super setUp];
+
+  self.tabBar = [[MDCTabBar alloc] init];
+  self.itemBar = [self.tabBar valueForKey:@"_itemBar"];
+  CGFloat tabBarHeight = [MDCTabBar defaultHeightForItemAppearance:self.tabBar.itemAppearance];
+  self.tabBar.frame = CGRectMake(0, 0, 200, tabBarHeight);
+  UITabBarItem *item1 = [[UITabBarItem alloc] initWithTitle:@"first" image:nil tag:0];
+  UITabBarItem *item2 = [[UITabBarItem alloc] initWithTitle:@"second" image:nil tag:0];
+  self.tabBar.items = @[ item1, item2 ];
+  [self.tabBar setNeedsLayout];
+  [self.tabBar layoutIfNeeded];
+}
+
+- (void)tearDown {
+  self.tabBar = nil;
+  self.itemBar = nil;
+
+  [super tearDown];
+}
+
+- (NSArray<MDCItemBarCell *> *)getItemBarCells {
+  NSMutableArray<MDCItemBarCell *> *arr = [[NSMutableArray alloc] init];
+  for (UITabBarItem *item in self.itemBar.items) {
+    NSInteger itemIndex = [self.itemBar.items indexOfObject:item];
+    if (itemIndex != NSNotFound) {
+      NSIndexPath *indexPath = [NSIndexPath indexPathForItem:itemIndex inSection:0];
+      UICollectionViewCell *cell = [self.itemBar.collectionView cellForItemAtIndexPath:indexPath];
+      if (cell) {
+        NSAssert([cell isKindOfClass:[MDCItemBarCell class]], @"All cells must be MDCItemBarCell");
+        MDCItemBarCell *itemCell = (MDCItemBarCell *)cell;
+        [arr addObject:itemCell];
+      }
+    }
+  }
+  return arr;
+}
+
+/**
+ Test to confirm behavior of initializing a @c MDCTabBar without any customization.
+ */
+- (void)testDefaultTabBarBehaviorWithInkEnabled {
```

Comment:
Similarly here as in https://github.com/material-components/material-components-ios/pull/7596#discussion_r293630962